components/humantask/org.wso2.carbon.humantask/src/main/java/org/wso2/carbon/humantask/core/scheduler/NotificationScheduler.java

```diff
@@ -0,0 +1,272 @@
+    private boolean isSMSNotification = false;
+    private boolean taskCompleted = false;
+    private static EmailEventAdapter emailAdapter;
+    private static SMSEventAdapter smsAdapter;
+    private NotificationScheduler notificationScheduler;
+    private ExecutorService exec;
+    private NotificationTask notificationTask;
+    private Map<String, String> emailDynamicProperties;
+    private Map<String, String> smsDynamicProperties;
+
+
+    private String prefix = "htd";
+    String mail_to = null;
+    //String ccEmail = null;
+    // String bccEmail = null;
+    String mail_subject = null;
+    String emailBody = null;
+    String emailBodyTagVal = null;
+    String smsReceiver = null;
+    String smsBodyTagVal = null;
+    String smsBody = null;
+
+
+    public void setExecutorService(ExecutorService executorService) {
+        exec = executorService;
+    }
+
+
+    public static synchronized EmailEventAdapter getEmailEventAdapter(Map<String, String> dynamicProperties) {
+        if (emailAdapter == null) {
+            emailAdapter = new EmailEventAdapter(null, dynamicProperties);
+        }
+        try {
+            emailAdapter.init();
+        } catch (OutputEventAdapterException e) {
+             log.error("unable to initialize adapter"+ emailAdapter + e);
+
+        }
+
+        return emailAdapter;
+    }
+
+    public static synchronized SMSEventAdapter getSMSEventAdapter(Map<String, String> dynamicProperties) {
+        if (smsAdapter == null) {
+            //Need staticProperties at init()
+
+            OutputEventAdapterConfiguration configuration = new OutputEventAdapterConfiguration();
+            configuration.setName("smsNotification");
+            configuration.setType("sms");
+            configuration.setStaticProperties(dynamicProperties);
+            configuration.setMessageFormat("text");
+
+            smsAdapter = new SMSEventAdapter(configuration, dynamicProperties);
+        }
+
+        try {
+            smsAdapter.init();
+        } catch (OutputEventAdapterException e) {
+            log.error("unable to initialize adapter"+ emailAdapter + e);
+
+        }
+
+        return smsAdapter;
+    }
+
+    public void checkForNotificationTasks(HumanTaskBaseConfiguration taskConfiguration, TaskCreationContext creationContext, TaskDAO task) {
+
+        isNotification = taskConfiguration.getConfigurationType().toString().equalsIgnoreCase("notification");
+
+        if (isNotification) {
+            isEmailNotification = HumanTaskServiceComponent.getHumanTaskServer()
+                    .getServerConfig().getEnableEMailNotification();
+            isSMSNotification = HumanTaskServiceComponent.getHumanTaskServer()
+                    .getServerConfig().getEnableSMSNotification();
+
+            executeNotifications(taskConfiguration, creationContext, task, isEmailNotification, isSMSNotification);
+
+        } else {
+            log.info("There are no type notification tasks");
```

Comment:
Updated log.info message and added java docs in commit https://github.com/wso2/carbon-business-process/commit/c037c2607734679028dc96c59761857176b6dfdb